sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/implementation/util/InternalContextN.java

```diff
@@ -71,12 +74,12 @@ public InternalContext put(Object key, Object value) {
 
     @Override
     public Object get(Object key) {
-        Object data = null;
+        Object data = SENTINEL;
 
         // Iterate in reverse order to get the most recent data first.
         for (int i = contexts.length - 1; i >= 0; i--) {
             data = contexts[i].get(key);
```

Comment:
Should we consider having `containsKey` method in InternalContext?

Reply:
We could, or we could have InternalContext use Optional whereas the public Context doesn't. I chose to go with the sentinel value as it is the cheapest computation wise.

Reply:
I'm ok with this.